### DISCUSSION TITLE
database directory location

### DISCUSSION THREAD

**Discussion by oldstager** [ArcadeData/arcadedb#161] [General]
I've managed to run server using:

$ bin/server.sh -Dserver.databaseDirectory=/some/path/data/arcadedb/

then I run console.sh and from there I create a db and import OpenBeer:

> create database beer
{beer}> IMPORT DATABASE https://github.com/ArcadeData/arcadedb-datasets/raw/main/orientdb/OpenBeer.gz.

I suspect, import ran successfully since I got this:

+---------+-----------------------------------------------------------------------------+
|NAME     |VALUE                                                                        |
+---------+-----------------------------------------------------------------------------+
|operation|import database                                                              |
|fromUrl  |https://github.com/ArcadeData/arcadedb-datasets/raw/main/orientdb/OpenBeer.gz|
|result   |OK                                                                           |
+---------+-----------------------------------------------------------------------------+

However, Studio doesn not show beer database and /some/path/data/arcadedb/ is empty. I still can use console.sh later and connect to beer database. 

I am still confuse, where are all those files located?

**Comment by lvca** [comment]
@oldstager you have to use `-Darcadedb.server.databaseDirectory=/some/path/data/arcadedb/`. Please let me know if works.

Check this: https://docs.arcadedb.com/#Settings/

**Reply by oldstager** [reply]
@lvca I finally found the databases, they are saved at $ARCADEDB_HOME/bin, eventhough I use -Darcadedb.server.databaseDirectory. Every newly created db is created at $ARCADEDB_HOME/bin (relative db path also created from $ARCADEDB_HOME/bin).

At first, I think this is because I use Fish, but changed ny shell to Bash didn't change anything.

**Comment by oldstager** [comment]
@lvca just checked using latest version (21.10.2). In studio, -Darcadedb.server.databaseDirectory works but doesn't work when using console.sh - create database command always create database in $ARCADEDB_HOME/bin. Still investigating.

**Reply by lvca** [reply]
@oldstager thanks for the report. Are you running the console with:

```
cd bin
./console.sh
```

Right?

Could you please try to run the console from ArcadeDB's root with:

```
bin/console.sh
```

**Reply by lvca** [reply]
Ok, I think I found the issue. Fixing it now...

**Comment by lvca** [comment]
Created issue https://github.com/ArcadeData/arcadedb/issues/173

**Reply by oldstager** [reply]
many thanks @lvca

**Comment by lvca** [comment]
Fixed. Please try with the `main` branch before we release the next 2021.11.1 in 1-2 weeks.

**Comment by mik1234mc** [comment]
I still cannot make it running, could check what is wrong on my side?

```
m@deb:~/projects/arcadedb/package/target/arcadedb-22.12.1.dir/arcadedb-22.12.1$ ./bin/server.sh -Darcadedb.server.databaseDirectory=/home/m/projects/arcadedb/package/target/pokdb
```
then in  ./bin/console.sh:
```
create database pok
```

Results: `/home/m/projects/arcadedb/package/target/pokdb` is empty and `pok` database is created in default location `~/projects/arcadedb/package/target/arcadedb-22.12.1.dir/arcadedb-22.12.1/databases`

**Comment by gramian** [comment]
AFAIK, the console always uses the default `databaseDirectory`, so you have to provide a relative path to the default path like:
```
create database ../../../../pokdb/pok
```
See also the discussion in https://github.com/ArcadeData/arcadedb/issues/704

**Comment by lvca** [comment]
I think it would be reasonable to allow the console to use the `arcadedb.server.databaseDirectory`, so you can use the console to create the database, and then launch the server with the same directory path.

**Reply by lvca** [reply]
Just created this issue: https://github.com/ArcadeData/arcadedb/issues/752

**Reply by lvca** [reply]
Ok, it's fixed. Please @mik1234mc let me know if the latest `main` branch works as expected.

**Reply by mik1234mc** [reply]
I got this error if I specify the databaseDirectory option:
```
arcadedb-23.1.1-SNAPSHOT$ ./bin/console.sh -Darcadedb.server.databaseDirectory=/home/db/data
Exception in thread "main" com.arcadedb.console.ConsoleException: No active database. Open a database first
	at com.arcadedb.console.Console.checkDatabaseIsOpen(Console.java:688)
	at com.arcadedb.console.Console.executeSQL(Console.java:509)
	at com.arcadedb.console.Console.execute(Console.java:215)
	at com.arcadedb.console.Console.parse(Console.java:599)
	at com.arcadedb.console.Console.main(Console.java:138)
```

**Reply by gramian** [reply]
I think the problem could lie in https://github.com/ArcadeData/arcadedb/blob/main/package/src/main/scripts/console.sh#L54 . I guess all commandline parameters are passed to console for parsing as console commands. A work around could be to use the `JAVA_OPTS` environment variable to pass the `databaseDirectory` setting.

**Reply by lvca** [reply]
@mik1234mc  Just pushed a fix for that. Now if you pass `-D<key>=<value>` is swallowed.
